From jengelh@linux01.gwdg.de  Mon May 21 17:35:05 2007
From: Jan Engelhardt <jengelh@linux01.gwdg.de>
Date: Thu, 17 May 2007 17:16:44 +0200 (MEST)
Subject: USB: Fix debug output of ark3116
To: Greg KH <greg@kroah.com>
Cc: <ark3116_driver@auctionant.de>
Message-ID: <Pine.LNX.4.61.0705171704230.3231@yvahk01.tjqt.qr>

From: Jan Engelhardt <jengelh@linux01.gwdg.de>

Fix debug output. Previously, it would output "0xFFFFFFB0" on 32-bit
archs (and probably "0xFFFFFFFFFFFFFFB0" on 64-bits), because buf is
taken as signed char, which is promoted to signed int, while %x always
expects an unsigned int.

Signed-off-by: Jan Engelhardt <jengelh@gmx.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/ark3116.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/drivers/usb/serial/ark3116.c
+++ b/drivers/usb/serial/ark3116.c
@@ -63,7 +63,8 @@ static inline void ARK3116_RCV(struct us
 				 request, requesttype, value, index,
 				 buf, 0x0000001, 1000);
 	if (result)
-		dbg("%03d < %d bytes [0x%02X]", seq, result, buf[0]);
+		dbg("%03d < %d bytes [0x%02X]", seq, result,
+		    ((unsigned char *)buf)[0]);
 	else
 		dbg("%03d < 0 bytes", seq);
 }
